### PR TITLE
HiOrg-Server OAuth client support

### PR DESCRIPTION
**Pull request by MartinBoehmer**
Added support for HiOrg-Server.

As long as [my pull request to scribejava](https://github.com/scribejava/scribejava/pull/867) has not been merged, you need to build and install my backport for this pull request to compile:
https://github.com/MartinBoehmer/scribejava/tree/hiorg-5.3.0

### CODE REVIEW

File: pac4j-oauth/src/main/java/org/pac4j/oauth/profile/hiorgserver/HiOrgServerProfileDefinition.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        final HiOrgServerProfile profile = newProfile();
+        final JsonNode json = JsonHelper.getFirstNode(body);
+        if (json != null) {
+            logger.debug("Extracting user profile from JSON node " + json);
+            profile.setId(ProfileHelper.sanitizeIdentifier(profile, JsonHelper.getElement(json, USER_ID)));
+            for (final String attribute : getPrimaryAttributes()) {
+                convertAndAdd(profile, PROFILE_ATTRIBUTE, attribute, JsonHelper.getElement(json, attribute));
+            }
+        } else {
+            logger.debug("No JSON node to extract user profile was created");
```
Comment (leleuj) [inline pac4j-oauth/src/main/java/org/pac4j/oauth/profile/hiorgserver/HiOrgServerProfileDefinition.java]: I don't think we want to build an empy profile, it seems to be a bad idea to me. Can you throw a `TechnicalException`instead?

File: pac4j-oauth/src/test/resources/logback-test.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<configuration>
```
Comment (leleuj) [inline pac4j-oauth/src/test/resources/logback-test.xml]: Outputting DEBUG logs for tests seems to be the right thing to do. Though, I have limit the logs produced by tests in order to have smaller logs file size in Travis. See: https://github.com/pac4j/pac4j/blob/master/pac4j-core/src/test/resources/logback.xml

In short, can you remove this file? Change the one in the core when you need more logs...

File: pac4j-oauth/src/main/java/org/pac4j/oauth/profile/hiorgserver/HiOrgServerProfileDefinition.java (hunk 1 above)
Comment (MartinBoehmer) [reply pac4j-oauth/src/main/java/org/pac4j/oauth/profile/hiorgserver/HiOrgServerProfileDefinition.java]: This is actually a behavior that all profiles have so far, i.e. none throws an exception. This is why I just added the log output. Throwing an exception makes a lot of sende, though. So, I agree to change that here, but it should be consitent across profiles.

File: pac4j-oauth/src/test/resources/logback-test.xml (hunk 2 above)
Comment (MartinBoehmer) [reply pac4j-oauth/src/test/resources/logback-test.xml]: Will do.

File: pac4j-oauth/src/main/java/org/pac4j/oauth/profile/hiorgserver/HiOrgServerProfileDefinition.java (hunk 1 above)
Comment (leleuj) [reply pac4j-oauth/src/main/java/org/pac4j/oauth/profile/hiorgserver/HiOrgServerProfileDefinition.java]: I'm in line. So let's change this one.

Added this issue for changing the others: https://github.com/pac4j/pac4j/issues/1175

**Comment by MartinBoehmer** [comment]
Okay, I think changed all the things you brought up.

**Comment by leleuj** [comment]
OK. Good. Let's wait for your PR to be accepted in scribejava and a new release to be cut...

**Comment by MartinBoehmer** [comment]
Scribejava 5.6.0 has been release and contains the required classes.
https://github.com/scribejava/scribejava/releases/tag/scribejava-5.6.0
However, switching to that version requires additional changes as some classes have been changed or removed from the API.

**Comment by leleuj** [comment]
Excellent! Keep me posted when the PR is ready for merge.

**Comment by MartinBoehmer** [comment]
The changes I meant are general compatibility fixes required for upgrading to the latest scribejava release with the HiOrg-Server API. I created a pull request for that: https://github.com/pac4j/pac4j/pull/1179

So if pac4j-oauth is upgraded to scribejava 5.6.0 this pull request can be merged.

**Comment by leleuj** [comment]
I just merged: https://github.com/pac4j/pac4j/pull/1179

Can you revert the `pom.xml` file and sync with the `release-notes.md` file? Thanks

**Comment by MartinBoehmer** [comment]
Okay, I think I'm done.

**Comment by leleuj** [comment]
Merging. Thanks for your contribution

**Pull request merged**